### PR TITLE
fix: fork should be rebased with upstream branch before MR

### PR DESCRIPTION
**Pull request by theflockers**
This PR adds the git_rebase function to git-functions so tasks that uses it can rebase the local (a fork, in case of file-updates) with the upstream with a given branch.

Also some small code adjusts.

### CODE REVIEW

File: utils/git-functions [hunk 1]
```diff
@@ -64,19 +63,56 @@ git_clone_and_checkout() {
...
     CLONE_URL="https://oauth2:${ACCESS_TOKEN}@${REPOSITORY#*://}"
-    git clone --depth 1 $CLONE_URL
-    cd $REPO_DIR
-    git checkout $REVISION
+    git clone --depth 1 "${CLONE_URL}"
+    cd "${REPO_DIR}"
+    git checkout "${REVISION}"
+}
+
+git_rebase() {
```
Comment (scoheb) [inline utils/git-functions:75]: where is this called?

File: utils/git-functions [hunk 2]
```diff
@@ -11,8 +11,8 @@ git_functions_usage() {
 }
 
 git_config_ident() {
-    git config user.name "${GIT_AUTHOR_NAME}" 
-    git config user.email "${GIT_AUTHOR_EMAIL}"
+    git config --global user.name "${GIT_AUTHOR_NAME}"
```
Comment (scoheb) [inline utils/git-functions:14]: why is this needed?

File: utils/git-functions (hunk 1 above)
Comment (theflockers) [reply utils/git-functions:75]: in the task. I will create a draft PR to show it.

**Review by scoheb — COMMENTED** [review]
some questions

File: utils/git-functions (hunk 2 above)
Comment (theflockers) [reply utils/git-functions:14]: two reasons: gitlab asks for the User e-mai when creating the MR, so if it is not passed it gives a prompt.

The second is because `git` asks  for the user as well to run the `git fetch`. I moved to `--global` because in some places `git_config_ident` is called there is no repo cloned yet, to it fails.

File: utils/git-functions (hunk 1 above)
Comment (theflockers) [reply utils/git-functions:75]: @scoheb  https://github.com/hacbs-release/app-interface-deployments/pull/46/files

**Review by scoheb — APPROVED** [review]

File: utils/git-functions (hunk 2 above)
Comment (johnbieren) [reply utils/git-functions:14]: I almost think we should set up global git configs in the container itself, but this is probably fine for the time being
